src/shared/psi-util.h

```diff
@@ -0,0 +1,30 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#pragma once
+
+#include <stdbool.h>
+
+#include "time-util.h"
+
+#define PRESSURE_FILE "/proc/pressure"
+
+typedef enum PressureType {
+        PRESSURE_TYPE_SOME = 0,
+        PRESSURE_TYPE_FULL = 1,
+} PressureType;
+
+typedef struct ResourcePressure {
+        float avg10;
+        float avg60;
+        float avg300;
```

Comment:
hmm, floats? are you sure? shouldn't this at least be "doubles"?

32bit floats are a really strange thing, they don't make too much sense all on their own i would say. they might make sense if you have many of them, but their inaccuracy makes them a bit more of a historic artifact than a real thing.

Does the kernel side use actual fp types for this? or do they use some integer with shifted decimal point or so (i.e. fixed point stuff)? If so, we should probably parse it in userspace the same time too